src/sdk/namespace_fs.js

```diff
@@ -293,11 +296,21 @@ const versions_dir_cache = new LRUCache({
                 usage += ent.name.length + 4;
             }
         }
-        return { time, stat, sorted_entries, usage };
+        return { time, stat, ver_dir_stat, sorted_entries, usage };
     },
-    validate: async ({ stat }, { dir_path, fs_context }) => {
+    validate: async ({ stat, ver_dir_stat }, { dir_path, fs_context }) => {
         const new_stat = await nb_native().fs.stat(fs_context, dir_path);
-        return (new_stat.ino === stat.ino && new_stat.mtimeNsBigint === stat.mtimeNsBigint);
+        if (ver_dir_stat) {
+            const versions_dir_path = path.normalize(path.join(dir_path, '/', HIDDEN_VERSIONS_PATH));
+            const new_versions_stat = await nb_native().fs.stat(fs_context, versions_dir_path);
+            return (new_stat.ino === stat.ino &&
+                    new_stat.mtimeNsBigint === stat.mtimeNsBigint &&
+                    new_versions_stat.ino === ver_dir_stat.ino &&
+                    new_versions_stat.mtimeNsBigint === ver_dir_stat.mtimeNsBigint);
+            } else {
+                return (new_stat.ino === stat.ino &&
+                        new_stat.mtimeNsBigint === stat.mtimeNsBigint);
+            }
```

Comment:
worth adding some tests

Reply:
sure

Reply:
I will add the test and update the patch asap. Some problems with the working system